src/components/debug-api.ts

```diff
@@ -0,0 +1,57 @@
+import inspector from 'inspector';
+import express, { Application, Request, Response } from 'express';
+import Logging from "./logging";
+
+const log = Logging.get("DebugAPI");
+
+export interface DebugApiOpts {
+    inspector?: {
+        port: number;
+        host?: string;
+    };
+    port: number;
+    host: string;
+}
+
+/**
+ * A HTTP based Debug API for bridging
+ */
+export class DebugAPI {
+    private app: Application;
+
+    constructor(private opts: DebugApiOpts) {
+        this.app = express();
+        this.app.post("/inspector/start", this.onInspectorStart);
+        this.app.post("/inspector/stop", this.onInspectorStop);
+    }
+
+    public async start(): Promise<unknown> {
+        return new Promise((res, rej) => {
+            this.app.once("error", rej);
+            this.app.listen(this.opts.port, this.opts.host, res);
+        });
+    }
+
+    public addRoute(method: "get"|"post"|"put"|"delete", path: string,
+        callback: (req: Request, res: Response) => void) {
+        this.app[method](path, callback);
+    }
+
+    private onInspectorStart(_: Request, res: Response) {
+        if (!this.opts.inspector) {
+            res.status(500).send({"error": "Inspector not configured"});
```

Comment:
An unconfigured feature shouldn't result in a HTTP 500 Server error.
How about 400 Bad request or 404 for pretending that the endpoint isn't there?